pytroll_collectors/tests/test_scisys.py

```diff
@@ -136,94 +194,159 @@ def test_reception(self):
 
         string = TwoMetMessage(input_stoprc)
         to_send = msg_rec.receive(string)
-        self.assertTrue(to_send is None)
-
-        filename = os.path.join('/tmp', viirs['uid'])
-        touch(filename)
-        string = TwoMetMessage(input_dispatch_viirs)
-        to_send = msg_rec.receive(string)
-        self.assertDictEqual(to_send, viirs)
-        os.remove(filename)
-
-        filename = os.path.join('/tmp', atms['uid'])
-        touch(filename)
-        string = TwoMetMessage(input_dispatch_atms)
-        to_send = msg_rec.receive(string)
-        self.assertDictEqual(to_send, atms)
-        os.remove(filename)
-
-        # NPP with start
+        assert to_send is None
+
+        with tempfile.TemporaryDirectory() as tmp:
+            filename = os.path.join(tmp, VIIRS['uid'])
+            touch(filename)
+            string = TwoMetMessage(create_input_dispatch_viirs_msg(tmp))
+            to_send = msg_rec.receive(string)
+            viirs_cpy = deepcopy(VIIRS)
+            viirs_cpy.pop('uri')
+            to_send.pop('uri')
+            TestCase().assertDictEqual(to_send, viirs_cpy)
```

Comment:
This structure is repeated a lot, so should be refactored.

Reply:
And each of these `with` branches should be a separate test case.

Reply:
Fixed. This can be further improved I am sure, but now it will be easier next time one is in for a re-feactoring I think.